src/main/java/edu/cmu/cs/cs214/rec10/plugin/SortWithSwapsPlugin.java

```diff
@@ -0,0 +1,118 @@
+package edu.cmu.cs.cs214.rec10.plugin;
+
+import edu.cmu.cs.cs214.rec10.framework.core.GameFramework;
+import edu.cmu.cs.cs214.rec10.framework.core.GamePlugin;
+
+import java.util.Random;
+
+public class SortWithSwapsPlugin implements GamePlugin {
+    private int ONE = 1;
+    private int TEN = 10;
+
+    // Describes whether the current move is over
+    public Boolean move = false;
+
+    // An array of the numbers 1-10
+    private int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+
+    // The game framework
+    private GameFramework framework = null;
+
+    // The last number clicked, which is to be swapped with the next number clicked
+    private int last = -1;
+
+    //The number of swaps the user has made
+    static Integer count = null;
+
+    public String getGameName() {
+        return new String("sort with swaps plugin");
+    }
+
+    public int getGridWidth() {
+        return TEN;
+    }
+
+    public int getGridHeight() {
+        return ONE;
+    }
+
+    public void onRegister(GameFramework framework) {
+        this.framework = framework;
+    }
+
+    public void onNewGame() {
+        // Reset count
+        count = 0;
+
+        // Make random permutation of the array
+        Random random = new Random();
+
+        int i;
+        // For every position i, chooses a random position >= i and swaps it with the value at position i,
+        // generating a random permutation of arr.
+        for(i = 0; i < 10; i++) {
+            int nextIndex;
+
+            // Find an index at position >= i to swap to position i.
+            while ((nextIndex = random.nextInt(10)) < i);  // Chooses a random number in the range [i, 10)
+
+            // Swap positions i, nextIndex
+            int temp = arr[i];
+            arr[i] = arr[nextIndex];
+            arr[nextIndex] = temp;
+        }
+
+        for(i = 0; i < 10; i++) {
+            framework.setSquare(i, 0, Integer.toString(arr[i]));
+        }
+    }
+
+    @Override
+    public void onNewMove() {
+        System.out.println("Status of game: at method onNewMove()");
```

Comment:
I feel like the status of the game should be more specific then just the method being called, because this doesn't give useful information to the user.